pkg/kubectl/cmd/logs.go

```diff
@@ -108,16 +114,23 @@ func NewCmdLogs(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().Bool("interactive", false, "If true, prompt the user for input when required.")
 	cmd.Flags().MarkDeprecated("interactive", "This flag is no longer respected and there is no replacement.")
 	cmdutil.AddInclude3rdPartyFlags(cmd)
+	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on")
 	return cmd
 }
 
 func (o *LogsOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
 	containerName := cmdutil.GetFlagString(cmd, "container")
+	selector := cmdutil.GetFlagString(cmd, "selector")
 	switch len(args) {
 	case 0:
-		return cmdutil.UsageError(cmd, logsUsageStr)
+		if selector == "" {
```

Comment:
Additionally, all the validation should happen in `Validate` method. The place where `ResourceArg` was previously checks is the place where you should be verifying between selector and resource name.